### PR TITLE
Branch xingyu dg contribution

### PR DESCRIPTION
**Pull request by Choo-Xing-Yu**
Fixed 
- 3.1.2 NAME and SUPPLIER to be NAME SUPPLIER
- Header -> remove all the item specific naming
- General features -> scrolling
- Add needs metric
- UG table

Added
- Editing, Find, Add separate with subheader item vs supplier
- 1 more FAQ
- Explanation for add's MAX_QUANTITY and TAG
- Glossary in UG
- UG add command add steps

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -28,16 +28,18 @@ as well as summary commands.
 
 4. Double-click the file to start the app. The similar GUI shown below (Figure 1) should appear in a few seconds. <br>
    *Note how the app contains some sample data but the installed version on your desktop might have a different data set.*<br>
```
Comment (Wincenttjoi) [inline docs/UserGuide.md]:
```suggestion
   *Note how the app contains some sample data but the installed version on your desktop might have a different data set.*
```

Comment (Wincenttjoi) [reply docs/UserGuide.md]: Not sure if this is better? Is the white spacing too big for every picture?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -28,16 +28,18 @@ as well as summary commands.
...
+
+    #<br>![Ui](images/UiWithAnnotation.png) <br>
+    Figure 1: Introduction to OneShelf User Interface
 
 5. Type the command `help start` in the command box and press Enter to execute it. <br>
    A new help window shown below (Figure 2) should appear on your desktop.
-   <br> ![Help Window](images/UiHelpStart.png) <br>
-   Figure 2: Help Start Window
+
+    #<br>![Help Window](images/UiHelpStart.png) <br>
```
Comment (Wincenttjoi) [inline docs/UserGuide.md]: The spacing for this seems the best thus far 
image

File: docs/UserGuide.md [hunk 3]
```diff
@@ -93,20 +99,54 @@ Figure 3: Help Summary Window
...
+
+2. Either press Enter on your keyboard or click Send to execute the command.
+
+3. Success Message will be shown in the Result Display (assuming there isn't an existing same item) and added item will show up in InventoryBook.
+<br>
+![add-i example one step three](images/add-i%20example%20one%20step%20three.PNG)
+<br>
+<br>
+
+Example 2: `add-i n/Tuna q/20 s/NTUC` using [Figure 1](#bruiimagesuiwithannotationpng-br) as a base.
```
Comment (Wincenttjoi) [inline docs/UserGuide.md]:
```suggestion
Example 2: `add-i n/Tuna q/20 s/NTUC` using [Figure 1](#bruiimagesuiwithannotationpng-br) as the starting point.
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -93,20 +99,54 @@ Figure 3: Help Summary Window
...
-An item is uniquely identified by its NAME and SUPPLIER.
+An item is uniquely identified by its `NAME` and `SUPPLIER`.
+* Note 1: `MAX_QUANTITY` denotes the ideal stock level of that particular item.
+* Note 2: `TAG` could be used to categorise items. EG: Duck can be tagged as meat. 
+
+Example 1: `add-i n/DUCK q/10 s/NTUC max/50 metric/KG` using [Figure 1](#bruiimagesuiwithannotationpng-br) as a base.
+
+1. Select the Command Box and type in `add-i n/DUCK q/10 s/NTUC max/50 metric/KG`.
+<br>
+![add-i example one step one](images/add-i%20example%20one%20step%20one.PNG)
```
Comment (Wincenttjoi) [inline docs/UserGuide.md:121]: Forgot to put Figure here

File: docs/UserGuide.md [hunk 5]
```diff
@@ -93,20 +99,54 @@ Figure 3: Help Summary Window
...
+<br>
+<br>
+
+2. Either press Enter on your keyboard or click Send to execute the command.
+
+3. Success Message will be shown in the Result Display (assuming there isn't an existing same item) and added item will show up in InventoryBook.
+<br>
+![add-i example one step three](images/add-i%20example%20one%20step%20three.PNG)
```
Comment (Wincenttjoi) [inline docs/UserGuide.md:129]: Another missing figure

File: docs/UserGuide.md [hunk 6]
```diff
@@ -93,20 +99,54 @@ Figure 3: Help Summary Window
...
+##### 3.1.2.a `add-i`
+
+Format: `add-i n/NAME q/QUANTITY [s/SUPPLIER] [max/MAX_QUANTITY] [metric/METRIC] [t/TAG]...​`
 * Adds `QUANTITY` from the current quantity of an inventory item if the item already exist.
-An item is uniquely identified by its NAME and SUPPLIER.
+An item is uniquely identified by its `NAME` and `SUPPLIER`.
+* Note 1: `MAX_QUANTITY` denotes the ideal stock level of that particular item.
+* Note 2: `TAG` could be used to categorise items. EG: Duck can be tagged as meat. 
+
+Example 1: `add-i n/DUCK q/10 s/NTUC max/50 metric/KG` using [Figure 1](#bruiimagesuiwithannotationpng-br) as a base.
```
Comment (Wincenttjoi) [inline docs/UserGuide.md]: If i were u i'll add multiple tags too for full set, then example 2 is minimum add = only name and quantity

File: docs/UserGuide.md [hunk 7]
```diff
@@ -275,9 +321,9 @@ OneShelf data are saved in the hard disk automatically after any command that ch
 
 
 
-#### 3.2.2 Scroll through command history
+#### 3.2.2 Scrolling through command history
 
-OneShelf commands are traversable much like Window's command prompt with the arrow up key traversing into previous commands and arrow down key traversing into next commands.
+OneShelf commands are traversable (up to 20 commands) much like Window's command prompt with the arrow up key traversing into previous commands and arrow down key traversing into next commands.
```
Comment (Wincenttjoi) [inline docs/UserGuide.md]: I think can be better rephrased with the action first. By pressing arrow up key ........ you can ........ What do you think?

**Comment by Wincenttjoi** [comment]
@xnoobftw you may want to consider orientating the reader on your screenshot. Lets say the screenshot is to explain you want to key in inside the commandbox, questions to consider:
1) Is it really necessary to screenshot the entire application for filling up the command box?
2) Would you want to highlight the commandbox since that is the focus?
3) Should you just screenshot the commandbox part instead?

Just feel that its a little bit overkill to repetitively screenshot, but just personal opinion.

File: docs/UserGuide.md (hunk 1 above)
Comment (Choo-Xing-Yu) [reply docs/UserGuide.md]: The white spacing is indeed abit too big with 2 <br>
image
added your change! Thanks!

File: docs/UserGuide.md (hunk 4 above)
Comment (Choo-Xing-Yu) [reply docs/UserGuide.md:121]: Figure is hyperlinked instead of placed there

File: docs/UserGuide.md (hunk 6 above)
Comment (Choo-Xing-Yu) [reply docs/UserGuide.md]: Got it!

**Comment by Choo-Xing-Yu** [comment]
> @xnoobftw you may want to consider orientating the reader on your screenshot. Lets say the screenshot is to explain you want to key in inside the commandbox, questions to consider:
> 
> 1. Is it really necessary to screenshot the entire application for filling up the command box?
> 2. Would you want to highlight the commandbox since that is the focus?
> 3. Should you just screenshot the commandbox part instead?
> 
> Just feel that its a little bit overkill to repetitively screenshot, but just personal opinion.

I think that screenshotting the entire applciation helps with navigating the app esp to a user.

To preserve consistency can i ask for your help in the highlighting the commandbox and result boxes respectively in both png? reason being we'd probably want it to look similar to Figure 1

File: docs/UserGuide.md (hunk 7 above)
Comment (Choo-Xing-Yu) [reply docs/UserGuide.md]: Paraphrased! Take a look!

**Review by Wincenttjoi — APPROVED** [review]

**Comment by Wincenttjoi** [comment]
LGTM

**Comment by Wincenttjoi** [comment]
> > @xnoobftw you may want to consider orientating the reader on your screenshot. Lets say the screenshot is to explain you want to key in inside the commandbox, questions to consider:
> > 
> > 1. Is it really necessary to screenshot the entire application for filling up the command box?
> > 2. Would you want to highlight the commandbox since that is the focus?
> > 3. Should you just screenshot the commandbox part instead?
> > 
> > Just feel that its a little bit overkill to repetitively screenshot, but just personal opinion.
> 
> I think that screenshotting the entire applciation helps with navigating the app esp to a user.
> 
> To preserve consistency can i ask for your help in the highlighting the commandbox and result boxes respectively in both png? reason being we'd probably want it to look similar to Figure 1

Remind me if i forgot to add in my PR. Also cos still need to wait for our last feature to be up

**Pull request merged**